Suggestions from CISA (#60)
vendorsec/mvsp: packages/mvsp/mvsp-alpha.en.asciidoc

MVSP Working Group <mvsp@groups.io>
Alpha - work in progress

Minimum Viable Secure Product (MVSP) is a list of essential application security controls that should be implemented in enterprise-ready products and services.

The controls are designed to be simple to implement and provide a good foundation for building secure and resilient systems and services.

We recommend that all companies building enterprise software and services, or otherwise handling sensitive information, implement the MVSP controls and, where possible, go well beyond them in their application security programs.

[cols="2,6a",stripes=none]
|===
2+<h| 1 Business controls
h| Control
h| Description

| 1.1 External vulnerability reports

> | * Publish a vulnerability disclosure policy on your website that outlines the testing scope, provides a legal safe harbor, and gives contact details for security reports

Reviewer: ```suggestion
| * Publish a vulnerability disclosure policy that outlines the testing scope, provides a legal safe harbor, and gives contact details for security reports
```

Previously we had "on your website" as part of this control, however pushback was that this was restrictive of vendors that may publish this through bug bounty services.

Reviewer: Accepting suggestion for now, we can review as required if there are any concerns.

h| Description

> | 1.1 Vulnerability reports

Reviewer: Should we consider renaming this control to `External vulnerability reports` to disambiguate this?

> | * Publish a vulnerability disclosure policy on your website, defining the method and/or point of contact for security reports. The policy should list the scope for testing and include legal safe harbor authorizing testing

Reviewer 2: ```suggestion
| * Publish a vulnerability disclosure policy on your website that outlines the testing scope, provides a legal safe harbor, and gives contact details for security reports
```

>   ** Example: Using template language from https://disclose.io[disclose.io] for a vulnerability disclosure policy, and the https://securitytxt.org[security.txt] standard to refer researchers to your policy

Reviewer 2: ```suggestion
  ** Example: Generate a policy with https://disclose.io[disclose.io] and publish it using https://securitytxt.org[RFC 9116].
```

Author: Since an organization would likely host a VDP on a separate page on their website and then include a link to it via security.txt, maybe the following language?

`** Example: Publish a policy generated with https://disclose.io[disclose.io] and refer to it using https://securitytxt.org[security.txt (RFC 9116)].`

Reviewer: I think the new suggestion makes sense here (recreated below for ease of applying).

```suggestion
  ** Example: Publish a policy generated with https://disclose.io[disclose.io] and refer to it using https://securitytxt.org[security.txt (RFC 9116)].
```

Note: We should ensure that the controls FAQ is updated here to cover definitions of `safe harbor`, and `disclosure policy` - separate PR

> * Develop and document procedures for triaging reported vulnerabilities, including a plan to and aim to eliminate identified classes of vulnerabilities in the long term

Reviewer 2: ```suggestion
* Develop and document procedures for triaging and remediating reported vulnerabilities
```

* Reasonable restrictions on testing can be defined

| 1.3 Self-assessment

> | Perform and publish security self-assessments using the latest MVSP release at least annually

Reviewer: ```suggestion
| Perform and make available security self-assessments using the latest MVSP release at least annually
```

We've seen pushback on specific language around publication in the past. Make available seems to be a reasonable mid-point for companies with a more closed culture of broad public sharing. This is in-line with the language in sub-processor control 4.3.

> | Perform and publish annual (at a minimum) security self-assessments using the latest MVSP release

Reviewer 2: ```suggestion
| Perform and publish security self-assessments using the latest MVSP release at least annually
```

| 1.4 External testing

> | Contract a security vendor to perform (at minimum) annual, comprehensive penetration tests of your products/services and systems they depend on

Reviewer 2: ```suggestion
| Contract a security vendor to perform comprehensive penetration tests of your products, services, and dependent systems at least annually
```

* Ensure data localization requirements are implemented in line with local regulations and contractual obligations

| 1.7 Incident handling

> | * Notify relevant parties about any security breach that affects sensitive information no later than 72 hours upon discovery, and upon learning any additional details of the breach. Consider reporting the breach to relevant national cybersecurity agencies

Reviewer: We may benefit from the reporting part being separate bullet point here. I adjusted the wording to make it read better stand-alone, but would appreciate other thoughts on wording here.

```suggestion
| * Notify relevant parties about any security breach that affects sensitive information no later than 72 hours upon discovery, and upon learning any additional details of the breach
* Consider reporting the breach to relevant national cybersecurity agencies in line with local guidance and regulations
```

Author: Looks good to me!

h| Description

| 2.1 Single Sign-On

> | Implement single sign-on using modern, maintained, and industry standard protocols and makes single sign-on available to configure for all customers at no additional cost

Reviewer 2: Suggestion:
> Implement single sign-on using modern, maintained, and industry standard protocols for all customers at no additional cost

> | Implement single sign-on using modern, maintained, and industry standard protocols and makes single sign-on available to configure for all customers at no additional cost

Reviewer 2: ```suggestion
| Implement single sign-on using modern, maintained, and industry standard protocols for all customers at no additional cost
```

  * Require the current password in addition to the new password during password change
  * Store passwords in a hashed and salted format using a memory-hard or CPU-hard one-way hash function
  * Enforce appropriate account lockout and brute-force protection on account access
  * Do not provide default passwords for users or administrators

| 2.5 Security libraries

> | Use modern, maintained, and industry standard frameworks, template languages, or libraries that systemically address implementation weaknesses by escaping the outputs and sanitizing the inputs

Reviewer: ```suggestion
| Use modern, maintained, and industry-standard frameworks, template languages, or libraries that systemically address implementation weaknesses by escaping the outputs and sanitizing the inputs
```

| 2.6 Dependency Patching
| * Ensure third-party dependencies are maintained and up-to-date, with security relevant updates having a severity score of "medium" or higher applied in line with your application patching schedule

>   * Upon becoming aware of a https://www.cisa.gov/known-exploited-vulnerabilities-catalog[Known Exploited Vulnerability] affecting a third-party dependency, patch the vulnerability as soon as possible, within seven days

Reviewer 2: ```suggestion
```
This is redundant with the previous requirement

Author: (for this and the following comment) -- the goal of these edits was to add some priority around patching particularly critical vulnerabilities, vs the same timeline for all vulnerabilities. The CISA KEV list can serve as a reference point for particularly critical vulnerabilities that have been exploited in the wild.

Reviewer: In this case it seems like the two points here are trying to achieve the same underlying result. I'm not sure that we can hope to assign a 7 day turn around as a minimum baseline however. I'd rather ensure that exploitability to considered as part of patch scheduling.

Perhaps a better edit would be:

```suggestion
  * Upon becoming aware of a https://www.cisa.gov/known-exploited-vulnerabilities-catalog[Known Exploited Vulnerability] affecting a third-party dependency, the patch should be prioritised
```

  * Application owner access to customer data (access transparency)

Logs must include user ID, IP address, valid timestamp, type of action performed, and object of this action.

> Logs must be stored for at least 30 days, and should not contain sensitive data or payloads. Logs should be made available at no additional charge

Reviewer 2: Suggestion:
> Logs must be stored for at least 30 days at no additional charge, and should not contain sensitive data or payloads.

| 2.8 Encryption

> | Use modern, maintained, and industry standard means of encryption to protect sensitive data in transit between systems, and at rest in online data storages and backups

Reviewer: ```suggestion
| Use modern, maintained, and industry-standard means of encryption to protect sensitive data in transit between systems, and at rest in online data storages and backups
```

  * Handling untrusted data. Example: Reusing data supplied by users within sensitive application contexts

| 3.4 Time to fix vulnerabilities
| * Produce and deploy patches to address application vulnerabilities that materially impact security within 90 days of discovery

>   * For vulnerabilities with evidence of active exploitation, produce and deploy patches within 14 days of discovery

Reviewer 2: In my view, although this control suggestion is logical, it introduces a sophisticated procedure for monitoring and pinpointing vulnerabilities designated as "actively exploited." This approach appears to deviate from the "minimality" principle of this guideline.

Reviewer: I'm not sure that I agree fully here. Effectively we're already asking companies to maintain visibility into patches for dependencies (this is already a large ask). Adding a single reference for known exploited vulnerabilities doesn't seem like a significant burden.

As above, I don't think that we can ask companies in a minimum baseline to patch within 14 days. We can recommend that known exploited vulnerabilities are prioritised.

>   * For vulnerabilities that require actions by customers, after patching publish a CVE record containing details of the vulnerability. The CVE should include the common weakness enumeration (CWE) field to identify the root cause

Reviewer 2: ```suggestion
  * Publish a security bulletin that details the vulnerability and its root cause if the remedy requires action from customers
```

Side note: maybe this should go under requirement 1.7?

Author: I think this makes more sense here since a vulnerability might not be an incident. Maybe this should be its own control (e.g. "3.6 Vulnerability Publishing")? If so, I'm happy to draft an FAQ.
